src/keystore.c

```diff
@@ -292,8 +297,15 @@ keystore_error_t keystore_unlock(const char* password, uint8_t* remaining_attemp
         return KEYSTORE_ERR_GENERIC;
     }
     if (password_correct) {
-        memcpy(_retained_seed, seed, sizeof(_retained_seed));
-        _is_unlocked_device = true;
+        if (_is_unlocked_device) {
+            // Already unlocked. Fail if the seed changed under our feet (should never happen).
```

Comment:
should we have an abort if it should never happen?

Reply:
Maybe. In this case, it was a bit easier to unit test, and the caller Aborts() anyway on a generic error.